### ISSUE TITLE
"kontena service ls" causes "internal server error"

### ISSUE THREAD

**Issue by vinhmmkh** [kontena/kontena#1741]
I will sum up my situation shortly below:

- I recently set up a Kontena grid **foo** using 4 nodes. 
- After that i setup a second grid **bar** using 1 node.
- The result is having 3 running docker container (kontena-server-haproxy/server-api/mongo-data on my master node. 
- I restartet the kontena-server service.
- Logged in locally  into my kontena grid **foo**  works fine.
- **But Using the command "kontena service ls" causes a "server internal error" Error.**

The first few lines of the error log logged in the server-api docker-container is: 

``` sh
NoMethodError (undefined method `to_path' for nil:NilClass
Did you mean?  to_a):
  /app/app/views/v1/grid_services/_grid_service.json.jbuilder:37:in `block (3 levels) in evaluate'
  /usr/lib/ruby/gems/2.3.0/gems/mongoid-4.0.2/lib/mongoid/relations/proxy.rb:150:in `map'
  /usr/lib/ruby/gems/2.3.0/gems/mongoid-4.0.2/lib/mongoid/relations/proxy.rb:150:in `method_missing'
  /usr/lib/ruby/gems/2.3.0/gems/mongoid-4.0.2/lib/mongoid/relations/embedded/many.rb:404:in `method_missing'
  /app/app/views/v1/grid_services/_grid_service.json.jbuilder:36:in `block (2 levels) in evaluate'
  /usr/lib/ruby/gems/2.3.0/gems/tilt-jbuilder-0.6.1/lib/tilt/jbuilder.rb:52:in `eval'
  /usr/lib/ruby/gems/2.3.0/gems/tilt-jbuilder-0.6.1/lib/tilt/jbuilder.rb:52:in `block in evaluate'
  /usr/lib/ruby/gems/2.3.0/gems/jbuilder-2.2.5/lib/jbuilder.rb:16:in `initialize'
  /usr/lib/ruby/gems/2.3.0/gems/tilt-jbuilder-0.6.1/lib/tilt/jbuilder.rb:8:in `initialize'
  /usr/lib/ruby/gems/2.3.0/gems/jbuilder-2.2.5/lib/jbuilder.rb:21:in `new'
  /usr/lib/ruby/gems/2.3.0/gems/jbuilder-2.2.5/lib/jbuilder.rb:21:in `encode'
...

```

Have you got any idea how to solve that problem? Without resetting all configs I made?

**Comment by SpComb** [comment]
Tracking a fix to what I assume is causing this in #1769

Can you confirm if this may have been caused by upgrading stacks to remove linked services, or could there also be some other cause?

**Comment by jnummelin** [comment]
The most probable root cause seems to be #1769, so I'll close this issue. Let's re-open if the root cause reveals to be something else.
